Remove usage of reflection in timeago-parser module

thigg opened the pull request:
We are trying to use the NewPipe extractor library for a native app for the mobile linux OS SailfishOS.
For this we intend to use graal's native-image.
When reflection is used, this needs an entry in a json file for each of the classes that may be used with reflection.

I created the put lines for the hashmap with a simple bashscript.
I think this should be totally equal (in terms of functionallity and maintainability) to the current solution without the drawback (for us) that it uses reflection.


- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [ ] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [ ] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

Stypox commented:
All of the java files under the `patterns/` folder were autogenerated by https://github.com/TeamNewPipe/NewPipeExtractor/blob/dev/timeago-parser/raw/java/GeneratePatternClasses.java, could you instead change that class to also autogenerate a java file which contains the hashmap, instead of manually hardcoding it?

dvag-thilo-kogge commented:
Sure, sounds fun

thigg commented:
@Stypox I updated the PR to generate the map with the generator.
While working at the generator I also moved it into its own gradle submodule, I hope this was okay?

thigg commented:
removed the line

Stypox commented on timeago-generator/src/main/java/com/github/TeamNewPipe/Utils.java:
```diff
@@ -1,3 +1,5 @@
+package com.github.TeamNewPipe;
```
The package name should be `org.schabi.newpipe.timeago_generator` I believe (also in the other `timeago-generator` classes).

Stypox commented on timeago-parser/src/main/java/org/schabi/newpipe/extractor/timeago/PatternMap.java:
```diff
@@ -0,0 +1,100 @@
+        patternMap.put("zu", zu.getInstance());
+    }
+
+    public static PatternsHolder getPattern(String languageCode) {
```
```suggestion
    public static PatternsHolder getPattern(final String languageCode) {
```

Stypox reviewed (approved):
Thank you! I tested importing the project in Android Studio and running the generator and everything worked without issues. I solved the two comments myself with two commits.

thigg commented:
Thank you. That was pleasant :)
